### ISSUE TITLE
Declaration of Abstract Class Properties 

### ISSUE THREAD

**Issue by micolec** [nus-cs2030/2223-s1#137]
# Summary
Was confused about whether we could declare abstract class properties as protected final. Usually in labs and mock PA the use of protected final will be flagged out on CodeCrunch (because we are supposed to declare all as private final). However in our notes, page 7/16 of lecture 4, it follows that abstract class properties should be declared as protected final. Clarified with my TA so i'm just sharing in case anyone else is curious! 

## Description 
My TA's reply: protected final is allowed. i think the confusion is for the mock pa and the prac pa where protected final is not allowed because in the prev sems only private final is allowed. But this sem prof relaxed it and allow the use of protected final. But as much as possible do use private final where possible 

## Screenshots (if any): 
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/86700792/193076209-2f69a32d-f62e-40f3-be2b-ca6805d12dc6.png">

**Comment by timotius-jason** [comment]
Hi,

I think protected final is allowed in abstract class. Even if it is not allowed, A quick workaround would be to declare it as private final, and create a protected getter. Hope this helps :D

**Comment by wilsonwid** [comment]
I agree with @timotius-jason. Since design is not graded at all, you can practically put getters for every `private final` attribute/field you put in.
